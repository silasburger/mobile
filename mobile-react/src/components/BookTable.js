import * as uuid from 'uuid';
import PropTypes from 'prop-types';

/**
 * A dumb presentational component that renders the book data
 * @returns HTML5 table element filled with book data
 */
const BookTable = ({books}) => {
    const headers = books.length ? Object.keys(books[0]) : [];

    const renderTableHeader = () => {
        return (
            <tr>
                {headers.map(header => {
                    return (<th key={uuid.v4()}>{header}</th>);
                })}
            </tr>
        )
    }

    const renderTableBody = () => {
        return books.map((book) => {
            return (
                <tr key={uuid.v4()}>
                    {headers.map(header => <td key={uuid.v4()}>{book[header]}</td>)}
                </tr>
            );
        });
    }


    return (
        <table>
            <thead>
                {renderTableHeader()}
            </thead>
            <tbody>
                {renderTableBody()}
            </tbody>
        </table>
    )
}

BookTable.propTypes = {
    books: PropTypes.array,
}

export default BookTable;