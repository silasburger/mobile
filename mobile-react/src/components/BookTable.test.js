import React from "react";
import { render, screen } from "@testing-library/react";
import BookTable from "./BookTable";

const books = [
    {
        id: 1,
        book_author: 'herr schmidt',
        book_title: 'requim for a dream',
        book_publication_year: 2001, 
        book_publication_country: 'United States',
        book_publication_city: 'Los Angeles', 
        book_pages: 203,
    },
    {
        id: 2,
        book_author: 'barack obama',
        book_title: 'dream child from hawaii',
        book_publication_year: 1962, 
        book_publication_country: 'Angola',
        book_publication_city: 'Luanda', 
        book_pages: 503,
    }
]

it("renders without crashing", function () {
  // this is a low-value test, but better than nothing
  render(<BookTable books={books} />);
});

it("has the correct data in table", function () {
    render(<BookTable books={books} />);
    expect(screen.getByText('herr schmidt')).toBeInTheDocument()
});


it("matches snapshot", function () {
  const { container } = render(<BookTable books={books} />);
  expect(container).toMatchSnapshot();
});

