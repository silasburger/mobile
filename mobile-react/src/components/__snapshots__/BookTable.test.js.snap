// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th>
          id
        </th>
        <th>
          book_author
        </th>
        <th>
          book_title
        </th>
        <th>
          book_publication_year
        </th>
        <th>
          book_publication_country
        </th>
        <th>
          book_publication_city
        </th>
        <th>
          book_pages
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          1
        </td>
        <td>
          herr schmidt
        </td>
        <td>
          requim for a dream
        </td>
        <td>
          2001
        </td>
        <td>
          United States
        </td>
        <td>
          Los Angeles
        </td>
        <td>
          203
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td>
          barack obama
        </td>
        <td>
          dream child from hawaii
        </td>
        <td>
          1962
        </td>
        <td>
          Angola
        </td>
        <td>
          Luanda
        </td>
        <td>
          503
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
